### PR TITLE
Add cloud2Polyline primitive distance calculation

### PR DESCRIPTION
**Pull request by smortvedt**
Similar to my last pull request, this pull request adds the distance to a polyline primitive calculation.  Note: Chris is working on the changes requested from #1015 specifically the error code enum and the assignments in the if statement.

### CODE REVIEW

File: CC/src/DistanceComputationTools.cpp [hunk 1]
```diff
@@ -2677,6 +2703,66 @@ int DistanceComputationTools::computeCloud2BoxEquation(GenericIndexedCloudPersis
...
+	}
+	if (!cloud->enableScalarField())
+	{
+		return -3;
+	}
+	ScalarType d = 0;
+	ScalarType dSumSq = 0;
+	for (unsigned i = 0; i < count; ++i)
+	{
+		ScalarType distSq = INFINITY;
```
Comment (smortvedt) [inline CC/src/DistanceComputationTools.cpp]: I wasn't sure what to use for infinity here

Comment (dgirardeau) [reply CC/src/DistanceComputationTools.cpp]: We normally use 'NAN_VALUE' (and you can use std::isnan to test it)

Comment (WargodHernandez) [reply CC/src/DistanceComputationTools.cpp]: I would argue that using NAN_VALUE is probably not the best choice here as the later use
```C++
distSq = std::min(distSq, computePoint2LineSegmentDistSquared(p, start, end));
```
Would become more complicated than required.

would it be acceptable to add the following to CCConst.h

```C++
//! INF as a ScalarType value
/** \warning: handle with care! **/
constexpr ScalarType INF_VALUE = std::numeric_limits<ScalarType>::infinity();

```
and then set any INF_VALUE at the end to NAN_VALUE if required outside of the inner loop?

if not then I would suggest we set the value to the first loop iteration value like:

```C++
ScalarType distSq = computePoint2LineSegmentDistSquared(p, 
							polyline->getPoint(0), 
							polyline->getPoint(1));
```

@dgirardeau what are your thoughts, is INF_VALUE a risk in other areas I'm not considering? Am I missing something that wouldn't require extra logic in the inner loop?

Comment (dgirardeau) [reply CC/src/DistanceComputationTools.cpp]: Oh if it's just a temporary value, I definitely prefer to init the distance the first time with the right value. The infinity value wouldn't be properly handled by the other components of CloudCompare (especially at display time, etc.).

File: qCC/mainwindow.cpp [hunk 2]
```diff
@@ -8997,6 +9003,17 @@ void MainWindow::doActionCloudPrimitiveDist()
 						ccConsole::Error(errString, "Box", returnCode);
 					break; 
 				}
+				case CC_TYPES::POLY_LINE:
+				{
+					signedDist = false;
```
Comment (WargodHernandez) [inline qCC/mainwindow.cpp]: Also need 
```C++
flippedNormals = false;
```

File: CC/src/DistanceComputationTools.cpp [hunk 3]
```diff
@@ -2151,6 +2152,31 @@ ScalarType DistanceComputationTools::computePoint2PlaneDistance(const CCVector3*
 	return static_cast<ScalarType>((CCVector3::vdot(P->u, planeEquation) - planeEquation[3])/*/CCVector3::vnorm(planeEquation)*/); //norm == 1.0!
 }
 
+ScalarType DistanceComputationTools::computePoint2LineSegmentDistSquared(const CCVector3* p, const CCVector3* start, const CCVector3* end)
+{
+	CCVector3 line = *end - *start;
+	CCVector3 vec;
+	ScalarType distSq;
+	PointCoordinateType t = line.dot(*p - *start);
+	t /= line.norm2();
```
Comment (dgirardeau) [inline CC/src/DistanceComputationTools.cpp]: maybe test the norm before using it as quotient? To avoid division by zero?

**Comment by WargodHernandez** [comment]
@dgirardeau Is there anything left you want fixed on this?

**Comment by dgirardeau** [comment]
Oh sorry, I was giving some times for others (well you or Andy ;) to make more comments.

**Comment by dgirardeau** [comment]
And thanks for the contribution ;)

**Pull request merged**